zrml/prediction-markets/src/lib.rs

```diff
@@ -1190,20 +1220,66 @@ mod pallet {
             let mut total_weight: Weight =
                 Self::process_subsidy_collecting_markets(now, T::MarketCommons::now());
 
+            // If we are at genesis or the first block the timestamp is be undefined. No
+            // market needs to be opened or closed on blocks #0 or #1, so we skip the
+            // evaluation. Without this check, new chains starting from genesis will hang up,
+            // since the loops in the `market_status_manager` calls below will run over an interval
+            // of 0 to the current time frame.
+            if now <= 1u32.into() {
+                return total_weight;
+            }
+
+            // We add one to the count, because `pallet-timestamp` sets the timestamp _after_
+            // `on_initialize` is called, so calling `now()` during `on_initialize` gives us
+            // the timestamp of the previous block.
+            let current_time_frame =
+                Self::calculate_time_frame_of_moment(T::MarketCommons::now()).saturating_add(1);
+
+            // On first pass, we use current_time - 1 to ensure that the chain doesn't try to
+            // check all time frames since epoch.
+            let last_time_frame =
+                LastTimeFrame::<T>::get().unwrap_or_else(|| current_time_frame.saturating_sub(1));
```

Comment:
`current_time_frame` will never be zero because we can assume that `pallet-timestamp` works correctly on block 2 or later. But if it were zero, then `last_time_frame` would be zero as well, and we would not handle any markets.

The linked line of code looks correct. Are you worried about something specific?

Reply:
OK for the first. About the linked code: Nevermind (I just wanted to make sure, that you look at it again).